src/bun.js/node/util/parse_args.zig

```diff
@@ -0,0 +1,687 @@
+            var short_option_str = short_option.toBunString(globalThis);
+            if (short_option_str.length() != 1) {
+                const err = JSC.toTypeError(JSC.Node.ErrorCode.ERR_INVALID_ARG_VALUE, "options.{s}.short must be a single character", .{option.long_name}, globalThis);
+                globalThis.vm().throwError(globalThis, err);
+                return error.ParseError;
+            }
+            option.short_name = short_option_str.charAtU8(0);
+        }
+
+        if (obj.getOwn(globalThis, "multiple")) |multiple_value| {
+            if (!multiple_value.isUndefined()) {
+                try validateBoolean(globalThis, multiple_value, "options.{s}.multiple", .{option.long_name});
+                option.multiple = multiple_value.toBoolean();
+            }
+        }
+
+        if (obj.getOwn(globalThis, "default")) |default_value| {
+            if (!default_value.isUndefined()) {
+                switch (option.type) {
+                    .string => {
+                        if (option.multiple) {
+                            try validateStringArray(globalThis, default_value, "options.{s}.default", .{option.long_name});
+                        } else {
+                            try validateString(globalThis, default_value, "options.{s}.default", .{option.long_name});
+                        }
+                    },
+                    .boolean => {
+                        if (option.multiple) {
+                            try validateBooleanArray(globalThis, default_value, "options.{s}.default", .{option.long_name});
+                        } else {
+                            try validateBoolean(globalThis, default_value, "options.{s}.default", .{option.long_name});
+                        }
+                    },
+                }
+                option.default_value = default_value;
+            }
+        }
+
+        log("[OptionDef] \"{s}\" (type={s}, short={s}, multiple={d}, default={?})", .{
+            String.init(long_option),
+            @tagName(option.type),
+            if (option.short_name) |chr| &[_]u8{chr} else "none",
+            @intFromBool(option.multiple),
+            option.default_value,
+        });
+
+        try option_definitions.append(option);
+    }
+}
+
+/// Process the args string-array and build an array identified tokens:
+/// - option (along with value, if any)
+/// - positional
+/// - option-terminator
+fn tokenizeArgs(globalThis: *JSGlobalObject, args: ArgsSlice, options: []const OptionDefinition, tokens: *std.ArrayList(Token)) !void {
+    var index: i32 = -1;
+    var group_count: i32 = 0;
+
+    // build a queue of args to process, because new args can be inserted during the processing
+    var queue_allocator = std.heap.stackFallback(32 * @sizeOf(JSValue), globalThis.allocator());
+    var queue = try std.ArrayList(JSValue).initCapacity(queue_allocator.get(), args.end - args.start);
```

Comment:
Actually for most part no JSValue are allocated, the only case is when a short option group is found (e.g. `-abc`) where it needs to create "virtual" arguments and push them into the queue.

I decided to use an array with JSValues, so I can reuse input values for the output as much as possible, to avoid doing: `input JSValue -> String -> JSValue for the output`